engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -908,7 +908,7 @@ private void execute() throws YarnException, IOException
           LOG.info("Releasing {} as resource with priority {} was already assigned", allocatedContainer.getId(), allocatedContainer.getPriority());
           releasedContainers.add(allocatedContainer.getId());
           numReleasedContainers++;
-          numRequestedContainers--;
+          numRequestedContainers++;    // undo the decrement above for this allocated container
```

Comment:
@sanjaypujare As this PR is a follow up for #474 and APEXCORE-624 that was supposed to address correct counting for `numRequestedContainers` and fix bugs that were always there, it was OK to fix the use case I provided as part of this PR.
I open APEXCORE-777 to specifically address the use case I provided.